src/routes/Root/Root.tsx

```diff
@@ -75,6 +118,24 @@ const Root = () => {
               </div>
             </NavLink>
           }
+          menuComponent={
+            <button
+              className="flex gap-4 justify-center"
+              aria-label="Avaa valikko"
+              onClick={() => {
+                setMegaMenuOpen(!megaMenuOpen);
```

Comment:
I think that when MegaMenu is opened, the user menu, and any other menus should be closed. Now they overlap 
![image](https://github.com/user-attachments/assets/f9477e12-80e5-4fa7-bb69-5b5e1284866f)